lib/consul/index.js

```diff
@@ -410,6 +410,16 @@ var process_records = function(branch, records, cb) {
 exports.getLastProcessedRef = function(branch, cb) {
   var key_name = create_key_name(branch, branch.name + '.ref', true);
   consul.kv.get({'key': key_name, token: token}, function(err, item) {
+    if(err) {
+      logger.warn('Cannot connect to consul to get the last processed reference id. Retrying...');
+        setTimeout(
+          function() {
+            exports.getLastProcessedRef(branch,cb);
+          },5000
+        );
+        // end this "thread"
+        return;
+    }
     /* istanbul ignore if */
     if (err) return cb(err);
```

Comment:
I think this is useless and should be removed as there should not be an err at this point.